src/In.ProjectEKA.DefaultHip/DataFlow/Collect.cs

```diff
@@ -21,19 +26,57 @@ public Collect(HiTypeDataMap hiTypeDataMap)
         public async Task<Option<Entries>> CollectData(DataRequest dataRequest)
         {
             var bundles = new List<Bundle> { };
-            var map = hiTypeDataMap.GetMap();
-            Log.Information($"Publishing dataRequest: {dataRequest}");
-            foreach (HiType hiType in dataRequest.HiType)
+            var results = FindPatientData(dataRequest);
+            foreach (var item in results)
             {
-                var dataList = map.GetValueOrDefault(hiType) ?? new List<string>();
-                foreach (var item in dataList)
-                {
-                    Log.Information($"Returning file: {item}");
-                    bundles.Add(await FileReader.ReadJsonAsync<Bundle>(item));
-                }
+                Log.Information($"Returning file: {item}");
+                bundles.Add(await FileReader.ReadJsonAsync<Bundle>(item));
             }
             var entries = new Entries(bundles);
             return Option.Some(entries);
         }
+
+        private bool WithinRange(HiDataRange range, DateTime date)
+        {
+            var fromDate = DateTime.ParseExact(range.From, "yyyy-MM-dd", null);
+            var toDate = DateTime.ParseExact(range.To, "yyyy-MM-dd", null);
+            return date >= fromDate && date < toDate;
+        }
+
+        public List<string> FindPatientData(DataRequest request) {
+            try
+            {
+                var jsonData = File.ReadAllText("demoPatientCareContextDataMap.json");
+                var patientDataMap = JsonConvert.DeserializeObject<Dictionary<string, Dictionary<string, List<PatientCCRecord>>>>(jsonData);
+
+                var listOfDataFiles = new List<string>();
+                foreach (var grantedContext in request.CareContexts)
+                {
+                    var refData = patientDataMap[grantedContext.PatientReferenceNumber];
+                    var ccData = refData?[grantedContext.CareContextReferenceNumber];
+                    if (ccData == null) continue;
+                    foreach (var ccRecord in ccData)
+                    {
+                        var captureTime = DateTime.ParseExact(ccRecord.capturedOn, "yyyy-MM-dd", null);
+                        if (!WithinRange(request.DataRange, captureTime)) continue;
+                        foreach (var hiType in request.HiType)
+                        {
+                            var hiTypeStr = hiType.ToString().ToLower();
+                            var dataFiles = ccRecord.data.GetValueOrDefault(hiTypeStr) ?? new List<string>();
+                            if (dataFiles.Count > 0)
+                            {
+                                listOfDataFiles.AddRange(dataFiles);
+                            }
+                        }
+                    }
+                }
+                return listOfDataFiles;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
```

Comment:
Use logger instead of using Console.WriteLine

Reply:
done